jsonrpc/client/zkevm.go

```diff
@@ -61,20 +61,21 @@ func (c *Client) BatchByNumber(ctx context.Context, number *big.Int) (*types.Bat
 // BatchesByNumbers returns batches from the current canonical chain by batch numbers. If the list is empty, the last
 // known batch is returned as a list.
 func (c *Client) BatchesByNumbers(_ context.Context, numbers []*big.Int) ([]*types.BatchData, error) {
-	var list []types.BatchNumber
+	batchNumbers := make([]string, 0, len(numbers))
 	for _, n := range numbers {
-		list = append(list, types.BatchNumber(n.Int64()))
+		bn := types.BatchNumber(n.Int64())
+		batchNumbers = append(batchNumbers, bn.StringOrHex())
 	}
-	if len(list) == 0 {
-		list = append(list, types.LatestBatchNumber)
+	if len(batchNumbers) == 0 {
+		latestBatchNum := types.LatestBatchNumber
+		batchNumbers = append(batchNumbers, latestBatchNum.StringOrHex())
 	}
 
-	var batchNumbers []string
-	for _, n := range list {
-		batchNumbers = append(batchNumbers, n.StringOrHex())
+	batchNumsMap := map[string][]string{
```

Comment:
Couldn't we just use that `types.BatchFilter` struct instead of this map?

Reply:
We can:
https://github.com/0xPolygon/cdk-validium-node/pull/111/commits/15efe728b0e1fd549e03edd960d947cb0962e63f